### ISSUE TITLE
Videos, Thumbnails, Icons Not Loading

### ISSUE THREAD

**Issue by ameenalasady** [reisxd/TizenTube#7]
Hi all,

I’m having trouble getting TizenTube to work on my Samsung TV. Here’s what I’ve done so far:

1. Enabled developer mode on my TV and set the host IP to my computer’s IP.
2. Installed Tizen Studio and the appropriate SDK version.
3. Connected my TV to Tizen Studio, which shows up in the Device Manager.
4. Created a certificate that works fine.
5. Loaded the TizenTube code onto Tizen Studio and changed the IP in index.html to my phone’s IP.
6. Ran both Launcher and TizenTube as Tizen Web Applications and installed them on my TV.
7. Changed the host IP on my TV to my phone’s IP and restarted the TV.
8. Downloaded the Android app, set the appid to the one shown in the Device Manager, and changed the IP to my TV’s IP.

At this point, everything seems to be working fine. I can run the server on the Android app and launch TizenTube on my TV. However, when TizenTube opens, nothing loads except for the video titles. There’s also a pop-up that says to press GREEN to open settings, but pressing GREEN on my remote control doesn’t do anything.

I’ve tried restarting everything and using the Node server instead, but I’m still having the same issue. Can anyone help me figure out what I’m doing wrong?

Thanks in advance!

EDIT: Trying the node server with debugging on gives the following logs:

![image](https://github.com/reisxd/TizenTube/assets/105679326/6286e5de-f282-4ecc-baa3-10a4f4786f8d)

I am not too sure why its giving the "config read failed" error.

Here is my config file:

![image](https://github.com/reisxd/TizenTube/assets/105679326/3888e54b-8724-40b2-90c2-3c2e2514a3df)

**Comment by reisxd** [comment]
> However, when TizenTube opens, nothing loads except for the video titles.

What's your TVs Tizen version/release year?

**Comment by ameenalasady** [comment]
The model number on the back of the TV says its the UN70TU6985F. It's released in 2021.

**Comment by reisxd** [comment]
> The model number on the back of the TV says its the UN70TU6985F. It was released in 2021. It is running on Tizen 6.0.

Maybe it was something wrong with YT itself? Could you try again? If not, I'll play around with the configuration of the app.

**Comment by ameenalasady** [comment]
I just tried again and I'm getting the same issue. The regular YouTube app works just fine on the same TV. Here's what I get: ![PXL_20230722_061007164.jpg](https://github.com/reisxd/TizenTube/assets/105679326/6b061150-011b-4d50-a7c6-edb79b35cd05)

**Comment by reisxd** [comment]
Could you run the app in debug mode (from your PC) and share the console logs? Note that you'll either need to use some workaround or an older version of Chrome/Chromium for the debugger.

**Comment by ameenalasady** [comment]
How can I debug the TizenTube app from Tizen Studio while also running the node server? The node server refuses to connect if sdb is running in the background. I tried debugging the app without having the node server running and here are the console logs (if they are of any use):


[www.youtube.com-1690012593136.log](https://github.com/reisxd/TizenTube/files/12135963/www.youtube.com-1690012593136.log)

**Comment by reisxd** [comment]
I don't really know if this is a Tizen issue or the underlying browser issue. But the browser tries to request from `about:blank` (which is weird due to the app redirecting to https://youtube.com/tv), which makes CORS unhappy.

**Comment by reisxd** [comment]
So, could you add `<tizen:content-security-policy>default-src *  data: blob: filesystem: about: ws: wss: 'unsafe-inline' 'unsafe-eval' 'unsafe-dynamic';  script-src * data: blob: 'unsafe-inline' 'unsafe-eval';  connect-src * data: blob: 'unsafe-inline';  img-src * data: blob: 'unsafe-inline';  frame-src * data: blob: ;  style-src * data: blob: 'unsafe-inline'; font-src * data: blob: 'unsafe-inline'; frame-ancestors * data: blob: 'unsafe-inline';</tizen:content-security-policy>` after [this line](https://github.com/reisxd/TizenTube/blob/main/apps/TizenTube/config.xml#L12)? This should allow for anything. Sure, it's vulnerable but it's for testing. If it still doesn't work, please give logs again.

**Comment by ameenalasady** [comment]
Hi @reisxd. Sorry for the late reply, I was in bed. Adding that line of code to config.xml does allow videos/thumbnails to load. However, opening the TizenTube app on my TV instantly crashes the node server with the following errors:
```
FetchError: request to http://192.168.2.103:34618/json failed, reason: connect ECONNREFUSED 192.168.2.103:34618
    at ClientRequest.<anonymous> (file:///D:/Downloads/Compressed/TizenTube-main/TizenTube-main/node_modules/node-fetch/src/index.js:108:11)
    at ClientRequest.emit (node:events:527:28)
    at Socket.socketErrorListener (node:_http_client:454:9)
    at Socket.emit (node:events:527:28)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  type: 'system',
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  erroredSysCall: 'connect'
  ```
  Once the node server crashes, the navigation on the app turns into a cursor (can not move left or right of the page). Any videos that are opened serve ads too (since the node server crashed).
  
  Trying to debug the app with the config.xml change is giving me problems. The dev tools very quickly disconnect, not giving me a chance to grab the logs. Here's the error that I get. THis pops up ~0.5 seconds after running the debugging tool.

**Comment by reisxd** [comment]
Well, that's weird. Could you increase the number [here](https://github.com/reisxd/TizenTube/blob/main/debuggerController.js#L10) to 10000 and try again?

**Comment by ameenalasady** [comment]
I still get cursor navigation on TizenTube and get the same error on the node server. Debugging also still does not work for some reason. Commenting out the line you mentioned in config.xml makes debugging work again.

**Comment by reisxd** [comment]
Could you try changing the config line I've told you to add to `<tizen:content-security-policy>font-src *;</tizen:content-security-policy>`? This should fix the icons not loading only. Also maybe the issue you're currently having.

**Comment by ameenalasady** [comment]
Same issue with that line. Cursor navigation on TizenTube with ads. Debugging also does not work. Node disconnects with the same error as previously.

**Comment by reisxd** [comment]
That is weird. I currently have no idea and I can't really test due to my TV being a Tizen 3 TV. 

I do have a theory though. Because we're setting CSP, Tizen automatically loads the app in a browser (which is probably why you see a cursor) and the debugger disconnects right after the browser is loaded. I have to somehow fix this without changing CSP then, I guess.

**Comment by ameenalasady** [comment]
I am pretty sure the app IS being loaded on the browser with the added config line. I can see the address bar of the browser very briefly when opening the app.

**Comment by JeoLeaf** [comment]
Hello, I am having the same issue with missing thumbnails and icons too. Running on Tizen 6.5
P.S. Thanks for this project, amazing work so far.

**Comment by reisxd** [comment]
@ameenalasady I don't really know what'll work if this doesn't. 

Could you change the origin parameter to `*` in [this line](https://github.com/reisxd/TizenTube/blob/main/apps/TizenTube/config.xml#L12)? Also, remove the CSP line if you haven't.

**Comment by ameenalasady** [comment]
@reisxd That fixed it! Everything works perfectly now. Is this considered a permanent fix or is it vulnerable? Thank you so much for your support. Very great work you have done.

**Comment by reisxd** [comment]
> @reisxd That fixed it! Everything works perfectly now. Is this considered a permanent fix or is it vulnerable? Thank you so much for your support. Very great work you have done.

I'll make a commit to fix it for others and close the issue then. I just made the app able to access any domain, which worked.

Older Tizens only needed the YT domain but now the newer ones require every domain I guess.

**Comment by reisxd** [comment]
Fixed in c9dda8cd5fd76d57baa0ad1753f073a122338fd3.
